Renamed Azure AD to Entra ID, Closes #5663

nicodecleyre opened the pull request:
Closes #5663

It seems there is a problem with the tests regarding `m365 login`, when running the tests for the command separately there are several tests that fail. If you perform the tests in their entirety for the entire CLI, there is no problem because somewhere in an AAD command an access token is simulated which remains until the tests of the `m365 login` command are performed.
Now that aad has been renamed to entra, the entra commands appear alphabetically after the login command (because login is in the commands folder), so I added a fix for the login command.

I also made all the links correctly and changed the naming from Azure AD to Entra ID in the files that already had to undergo changes.

milanholemans commented:
Thank you @nicodecleyre, we'll try to review it ASAP!

martinlingstuyl commented:
@Jwaegebaert, we will be running into SEO issues and broken links when we totally change the url structure of the docs to entra. Would you know if we can use alias-URLs in docusaurus on page level? Or would you know another way to deal with this? @pnp/cli-for-microsoft-365-maintainers

milanholemans commented:
Does it matter that the links break @martinlingstuyl? With other renamings, we also just didn't support the old URL anymore.

martinlingstuyl commented:
> With other renamings, we also just didn't support the old URL anymore.

In general I think it does matter that a link breaks. Especially if it's such a big section of commands. Blog posts from all over the place that reference our site are going to have trouble with this, which is unfortunate, especially if we would be able to solve it with ease.

milanholemans commented:
I can only think of a [client-side redirect plugin](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects) (not sure if this is supported OOTB). But I fear this will become a tangle of redirects after a while (last year we did a few renames already).

martinlingstuyl commented on src/m365/entra/commands/group/group-get.ts:
```diff
@@ -15,13 +16,17 @@ interface Options extends GlobalOptions {
 
   public get description(): string {
-    return 'Gets information about the specified Azure AD Group';
+    return 'Gets information about the specified Entra ID Group';
```
```suggestion
    return 'Gets information about the specified Entra group';
```

martinlingstuyl commented on src/m365/entra/commands/group/group-remove.ts:
```diff
@@ -17,13 +18,17 @@ interface Options extends GlobalOptions {
 
   public get description(): string {
-    return 'Removes an Azure AD group';
+    return 'Removes an Entra ID group';
```
```suggestion
    return 'Removes an Entra group';
```

martinlingstuyl commented on src/m365/entra/commands/group/group-user-list.ts:
```diff
@@ -24,13 +25,17 @@ interface ExtendedUser extends User {
 
   public get description(): string {
-    return 'Lists users of a specific Azure AD group';
+    return 'Lists users of a specific Entra ID group';
```
```suggestion
    return 'Lists users of a specific Entra group';
```

martinlingstuyl commented on src/m365/entra/commands/groupsetting/groupsetting-list.ts:
```diff
@@ -3,14 +3,19 @@ import { Logger } from '../../../../cli/Logger.js';
 
   public get description(): string {
-    return 'Lists Azure AD group settings';
+    return 'Lists Entra ID group settings';
```
```suggestion
    return 'Lists Entra group settings';
```

martinlingstuyl commented on src/m365/entra/commands/groupsettingtemplate/groupsettingtemplate-get.ts:
```diff
@@ -15,13 +16,17 @@ interface Options extends GlobalOptions {
 
   public get description(): string {
-    return 'Gets information about the specified Azure AD group settings template';
+    return 'Gets information about the specified Entra ID group settings template';
```
```suggestion
    return 'Gets information about the specified Entra group settings template';
```

martinlingstuyl commented on src/m365/entra/commands/groupsettingtemplate/groupsettingtemplate-list.ts:
```diff
@@ -3,14 +3,19 @@ import { Logger } from '../../../../cli/Logger.js';
 
   public get description(): string {
-    return 'Lists Azure AD group settings templates';
+    return 'Lists Entra ID group settings templates';
```
```suggestion
    return 'Lists Entra group settings templates';
```

martinlingstuyl reviewed (changes requested):
Hi @nicodecleyre, great start! A big PR cannot go without a big list of comments: let's fix these first before we merge..

nicodecleyre commented on docs/src/config/sidebars.ts:
```diff
@@ -360,22 +360,22 @@ const sidebars: SidebarsConfig = {
             {
               type: 'doc',
               label: 'siteclassification disable',
-              id: 'cmd/aad/siteclassification/siteclassification-disable'
```
Can this be a solution? (cc @Jwaegebaert)
https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects

Jwaegebaert replied:
This plugin should indeed work. @milanholemans also mentioned it before. We'll be tracking it in a separate issue: #5738

martinlingstuyl replied:
@nicodecleyre, so you can ignore this comment, we'll be implementing the redirect separately. I'll resolve this one.

martinlingstuyl commented:
> I can only think of a [client-side redirect plugin](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects) (not sure if this is supported OOTB). But I fear this will become a tangle of redirects after a while (last year we did a few renames already).

OK, so we're fixing this in a separate PR @nicodecleyre, you don't have to implement this.

nicodecleyre commented:
Hi @martinlingstuyl, thank you for your time in reviewing this big PR 🤙. Do you know how I can mark these files as moved? 

I tried to reset the files, move them with `git mv` and made the changes separately, but git keeps on marking them as deleted and added. Although it recognises it as a 'move and change' with `git log --stat`
![image](https://github.com/pnp/cli-microsoft365/assets/35696168/51bdfd60-b46c-42ea-994e-95448fa3cc77)

martinlingstuyl commented:
Hi @nicodecleyre, that's annoying. It may have something to do with the amount of changes in the file, and whether git still sees it as the same file. 
It should work if you do it in separate commits. So first use git mv and commit the whole shebang, and then edit the new file with the changes necessary. 

The only thing I hope is that this will stay correctly marked as soon as we fixup the changes into a single commit.

nicodecleyre commented:
Just for the record, i started a new fresh branch from main, moved only the 6 files with `git mv`, commited, then did the changes, commited, and also there it marks these files eventually as deleted and added 😞 

https://github.com/pnp/cli-microsoft365/compare/main...nicodecleyre:cli-microsoft365:aad-to-entra-part-2

martinlingstuyl commented:
Alas, that's what I feared... well in that case it is as it is. We'll just have to live with it. You may resolve my comment

martinlingstuyl commented on docs/docs/v3-upgrade-guidance.mdx:
```diff
@@ -38,15 +38,15 @@ After upgrading to v3, `@pnp/cli-microsoft365`, any scripts that use the `o365`
 ***What action do I need to take?***
 
-After upgrading to v3, `@pnp/cli-microsoft365`, any scripts that make use of a custom Azure AD identity for authentication will need to update the setting of the environment variables to use the new names, `CLIMICROSOFT365_AADAPPID` and `CLIMICROSOFT365_TENANT`.
+After upgrading to v3, `@pnp/cli-microsoft365`, any scripts that make use of a custom Entra ID for authentication will need to update the setting of the environment variables to use the new names, `CLIMICROSOFT365_AADAPPID` and `CLIMICROSOFT365_TENANT`.
```
```suggestion
After upgrading to v3, `@pnp/cli-microsoft365`, any scripts that make use of a custom Entra identity for authentication will need to update the setting of the environment variables to use the new names, `CLIMICROSOFT365_AADAPPID` and `CLIMICROSOFT365_TENANT`.
```

martinlingstuyl commented on docs/docs/v3-upgrade-guidance.mdx:
```diff
@@ -38,15 +38,15 @@ After upgrading to v3, `@pnp/cli-microsoft365`, any scripts that use the `o365`
+In Office 365 CLI v2, we provided two environment variables, `OFFICE365CLI_AADAPPID` and `OFFICE365CLI_TENANT`. These environment variable could be updated to point the CLI to your own custom Entra identity to use for logging into your Microsoft 365 tenant instead of using the multi-tenant PnP Management Shell identity used as the default identity.
 
-Find out more about this feature: [Using your own Azure AD identity](user-guide/using-own-identity.mdx)
+Find out more about this feature: [Using your own Entra ID](user-guide/using-own-identity.mdx)
```
```suggestion
Find out more about this feature: [Using your own Entra identity](user-guide/using-own-identity.mdx)
```

martinlingstuyl reviewed (approved):
Looks fine, I'll merge it ASAP!

martinlingstuyl commented:
Merged manually, thank you!! 🎆
